browser/page_mapping.go

```diff
@@ -423,70 +423,87 @@ func mapPage(vu moduleVU, p *common.Page) mapping { //nolint:gocognit,cyclop
 func mapPageOn(vu moduleVU, p *common.Page) func(common.PageOnEventName, sobek.Callable) error { //nolint:funlen
 	rt := vu.Runtime()
 
+	pageOnEvents := map[common.PageOnEventName]struct {
+		mapp func(vu moduleVU, event common.PageOnEvent) mapping
+		prep func() error
```

Comment:
Might be worth adding a comment here. Maybe something like:

```suggestion
		prep func() error // If set, this will run before the handler itself.
```